core/src/test/java/com/linecorp/armeria/client/endpoint/StickyEndpointSelectionStrategyTest.java

```diff
@@ -99,9 +100,8 @@ public void select() {
 
     private static ClientRequestContext contextWithHeader(String k, String v) {
         ClientRequestContext ctx = mock(ClientRequestContext.class);
-        HttpRequest testReq = new DefaultHttpRequest();
-        testReq.headers().set(AsciiString.of(k), v);
-        when(ctx.request()).thenReturn(testReq);
+        when(ctx.request()).thenReturn(HttpRequest.of(HttpHeaders.of(HttpMethod.GET, "/")
+                                                                 .set(AsciiString.of(k), v)));
```

Comment:
I'm curious if it is a good idea to provide a method like `HttpHeaders.toHttpRequest()` and `HttpHeaders.toHttpResponse()`. This will give a user two ways to create an `HttpRequest` or `HttpResponse`, which may be good or bad. What do you think? (We recently deprecated `AggregatedHttpMessage.toHttp*()` in favor of `Http*.of(..)` by the way, but wanted to know which a user would prefer.)

Reply:
It's a difficult question, especially since the number of characters is exactly the same ;) I like the reduced cognitive load of having only one entrypoint to HttpRequest/Response. Since the buck stops at the `toHttpRequest` in general, I don't think there's particular value to being fluent either. But it's a personal preference so wonder what others think.

Reply:
Probably not the scope of this PR. Let us keep this in mind though. :-)